parser/js/src/test/scala/hmda/parser/fi/DateGenerators.scala

```diff
@@ -0,0 +1,15 @@
+package hmda.parser.fi
+
+import org.scalacheck.Gen
+import scalajs.js
+
+object DateGenerators {
+  def randomDate: Gen[Int] = {
+    val beginDate = new js.Date("2017-01-01")
+    val endDate = new js.Date("2020-12-31")
+
+    for {
+      randomDate <- Gen.choose(beginDate.getTime(), endDate.getTime())
+    } yield new js.Date(randomDate).getTime().toInt
```

Comment:
The `JVM` version produces an `Int` of a specific date format.  There, a `Date` object is turned into a formatted `String` (`"yyyyMMdd"`), then converted that to an `Int` (e.g. `20161102`).

The `JS` version is producing a random Date in milliseconds, which have a different length than the `JVM` output.